### PR TITLE
Convert require to require_relative

### PR DESCRIPTION
**Pull request by lamont-granquist**
This gives a speed boost since rubygems does not have to scan through
every gem in the gemset in order to find the file.

Signed-off-by: Lamont Granquist <lamont@scriptkiddie.org>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
On my mac this gives a 16% speed boost:

```
/opt/chef/bin/chef-client --version  1.46s user 0.57s system 99% cpu 2.050 total
/opt/chef/bin/chef-client --version  1.32s user 0.42s system 99% cpu 1.754 total
```

Conversion was done with this script:

https://gist.github.com/d44421dea302eff20c1cb3e1552e8c19

File: lib/chef/application/client.rb [hunk 1]
```diff
@@ -17,19 +17,19 @@
...
-require "chef/application"
-require "chef/client"
-require "chef/config"
-require "chef/daemon"
-require "chef/log"
-require "chef/config_fetcher"
-require "chef/handler/error_report"
-require "chef/workstation_config_loader"
-require "chef/mixin/shell_out"
+require_relative ""
```
Comment (tas50) [inline lib/chef/application/client.rb]: Does this just do the current dir?

Comment (lamont-granquist) [reply lib/chef/application/client.rb]: WTF?  no, that's a bug.

Comment (lamont-granquist) [reply lib/chef/application/client.rb]: weird that all those mangled lines didn't cause any test failures...

**Comment by lamont-granquist** [comment]
Unfortunately we're still slower than Chef-14 due to appbundling changes:

```
# time /opt/chef/bin/chef-client --version
Chef: 14.12.9
/opt/chef/bin/chef-client --version  0.97s user 0.27s system 98% cpu 1.255 total
```

**Pull request merged**